src/Servers/Kestrel/test/FunctionalTests/RequestTests.cs

```diff
@@ -509,7 +506,7 @@ public async Task AbortingTheConnectionSendsFIN()
         [MemberData(nameof(ConnectionAdapterData))]
         public async Task ConnectionClosedTokenFiresOnClientFIN(ListenOptions listenOptions)
         {
-            var testContext = new TestServiceContext(LoggerFactory);
+            var testContext = new TestServiceContext(LoggerFactory) { ExpectedConnectionMiddlewareCount = listenOptions._middleware.Count };
```

Comment:
Should we remove asserting the middleware count? I feel like it's causing a lot of code churn with very little value.

Reply:
I think we should remove it, but let @halter73 decide, he's the one that added it.

Reply:
I'm fine with removing it. Before it was very unusual to have multiple, so it almost always signified a test bug where a ListenOptions object was being reused.

Now that so many tests add non-HttpConnectionMiddleware, I'm fine with removing it. Just watch out whenever you're writing kestrel tests not to reuse ListenOptions.